frontend/src/utils/firebaseHelpers.js

```diff
@@ -12,86 +11,68 @@ const firebaseConfig = {
-  const messaging = getMessaging(app);
+export const getFCMRegistrationToken = async (messaging) => {
+  // Get registration token. Initially this makes a network call, once retrieved
+  // subsequent calls to getToken will return from cache.
+  if (!messaging) return;
 
-  requestPermission();
-  getFCMRegistrationToken(messaging, 'web', false);
-};
+  try {
+    const registrationToken = await getToken(messaging, {
+      vapidKey: process.env.REACT_APP_FIREBASE_VAPID_KEY
+    });
+
+    if (!registrationToken) return;
 
-const requestPermission = () => {
-  console.log('Request permission...');
+    activateDevice(registrationToken);
+    return registrationToken;
+  } catch (err) {
+    console.log('An error occurred while retrieving token. ', err);
+  }
+};
 
-  Notification.requestPermission().then(async (permission) => {
-    if (permission === 'granted') {
-      console.log('Notification permission granted.');
+const activateDevice = (token) => {
+  if (!token) return;
 
-      getFCMRegistrationToken();
-    }
+  axios.post('/devices/', {
+    type: 'web',
+    registration_id: token,
+    active: true
   });
 };
 
-const getFCMRegistrationToken = (messaging, type = 'web', active = true) => {
-  // Get registration token. Initially this makes a network call, once retrieved
-  // subsequent calls to getToken will return from cache.
-  if (!messaging) {
-    return;
-  }
+export const deactivateFirebaseDevice = (token) => {
+  if (!token) return;
 
-  getToken(messaging, {
-    vapidKey: process.env.REACT_APP_FIREBASE_VAPID_KEY
-  })
-    .then((registrationToken) => {
-      if (registrationToken) {
-        console.log(
-          `%c테스트를 위해 다음의 등록키를 backend/adoorback/feed/views.py의 registration_token에 등록해보세요!: ${registrationToken}`,
-          'color: yellow'
-        );
-        // Send the token to your server and update the UI if necessary
-        axios.post('/devices/', {
-          type,
-          registration_id: registrationToken,
-          active
-        });
-      } else {
-        // Show permission request UI
-        console.log(
-          'No registration token available. Request permission to generate one.'
-        );
-      }
-    })
-    .catch((err) => {
-      console.log('An error occurred while retrieving token. ', err);
-    });
+  axios.post('/devices/', {
+    type: 'web',
+    registration_id: token,
+    active: false
+  });
 };
 
-const addForegroundMessageEventListener = (messaging) => {
+export const addForegroundMessageEventListener = (messaging, onClick) => {
   onMessage(messaging, (payload) => {
-    console.log('Received foreground message', payload);
-
-    // TODO: 노티의 내용 구성, 제목, 링크, 아이콘 등의 설정 필요
+    // TODO: 노티 아이콘 설정
     const {
       notification: { title, body },
       data: { url }
     } = payload;
 
     const noti = new Notification(title, { body });
+
     noti.onclick = () => {
-      // TODO: 노티를 클릭했을 때 동작 정의 필요 e.g. URL 이동 등
-      console.log(url);
+      onClick(url);
```

Comment:
data에 항상 url이 존재하는 상황에서만 동작해야할 것 같아요!

```suggestion
      url && onClick(url);
```
정도로..? 나중에 url이 없거나 빈 스트링인 경우가 분명 생길 수 있을 것 같고, 그때를 대비해서 지금 방어 로직을 넣어두는게 나쁘지 않을 것 같아서요!

Reply:
오오 꼼꼼한 리뷰 감사합니다ㅎㅎ
url 있는 경우만 실행하도록 수정했어요([af630ef](https://github.com/GooJinSun/diivers/pull/150/commits/af630efdfbc75b2dedf80a0b81823b93e154ac7a))
타입스크립트 얼른 도입해야겠네요..! 생각난김에 develop->main 빨리 머지할게요ㅠㅠ! @koyrkr